packages/@aws-cdk/aws-cognito/lib/user-pool-security.ts

```diff
@@ -0,0 +1,247 @@
+  /**
+   * The email address that is sending the email.
+   * @default - Not set
+   */
+  readonly from?: string;
+  /**
+   * The MFA email template used when MFA is challenged as part of a detected risk.
+   * @default - Not set
+   */
+  readonly mfaEmail?: NotifyEmail;
+  /**
+   * The email template used when a detected risk event is allowed.
+   * @default - Not set
+   */
+  readonly noActionEmail?: NotifyEmail;
+  /**
+   * The destination to which the receiver of an email should reply to.
+   * @default - Not set
+   */
+  readonly replyTo?: string;
+  /**
+   * Email template used when a detected risk event is blocked.
+   */
+  readonly sourceArn: string;
+}
+
+/**
+ * The configuration for actions and notification for different levels of risk detected for a potential account takeover.
+ */
+export interface AccountTakeoverRiskConfiguration {
+  /**
+   * The actions of account takeover risk configuration.
+   */
+  readonly actions: AccountTakeoverActions;
+
+  /**
+   * The notify configuration used to construct email notifications.
+   * @default - no configured
+   */
+  readonly notifyConfiguration?: NotifyConfiguration;
+}
+
+/**
+ * The action for compromised credentials.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEventAction {
+  /** Blocking requires users to choose another password */
+  BLOCK = 'BLOCK',
+  /** all attempted uses of compromised credentials to Amazon CloudWatch */
+  NO_ACTION = 'NO_ACTION',
+}
+
+/**
+ * The actions for compromised credentials.
+ */
+export interface CompromisedCredentialsActions {
+  /**
+   * The action compromised credentials event.
+   */
+  readonly eventAction: CompromisedCredentialsEventAction;
+}
+
+/**
+ * The event type for compromised credentials check.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEvent {
+  /** Users sign in */
+  SIGN_IN = 'SIGN_IN',
+  /** Users change password */
+  PASSWORD_CHANGE = 'PASSWORD_CHANGE',
+  /** Users sign up */
+  SIGN_UP = 'SIGN_UP',
+}
+
+/**
+ * The configuration for compromised credentials risk.
+ */
+export interface CompromisedCredentialsRiskConfiguration {
+  /**
+   * The actions for compromised credentials risk.
+   */
+  readonly actions: CompromisedCredentialsActions;
+  /**
+   * The trigger for compromised credentials check.
+   * @default - Not set.
+   */
+  readonly eventFilter?: CompromisedCredentialsEvent[];
+}
+
+/**
+ * The configuration to override the risk decision.
+ */
+export interface RiskExceptionConfiguration {
+  /**
+   * Overrides the risk decision to always block the pre-authentication requests
+   */
+  readonly blockedIpRangeList?: string[];
+  /**
+   * Risk detection is not performed on the IP addresses in the range list
+   */
+  readonly skippedIpRangeList?: string[];
```

Comment:
Rename these to `alwaysBlockIpCidrs` and `alwaysAllowIpCidrs`. Call out in their documentation that these have to be in the CIDR format and provide a couple of examples.